src/RealtimeEffectPanel.cpp

```diff
@@ -987,6 +971,35 @@ RealtimeEffectPanel::RealtimeEffectPanel(wxWindow* parent, wxWindowID id, const
    mEffectList = effectList;
 
    SetSizerAndFit(vSizer.release());
+
+   mTrackListChanged = TrackList::Get(mProject).Subscribe([this](const TrackListEvent& evt) {
+         auto track = evt.mpTrack.lock();
+         auto waveTrack = dynamic_cast<WaveTrack*>(track.get());
+
+         if (waveTrack == nullptr)
+            return;
+         
+         switch (evt.mType)
+         {
+         case TrackListEvent::TRACK_DATA_CHANGE:
+            if (mCurrentTrack.lock() == track)
+               mTrackTitle->SetLabel(track->GetName());
+            UpdateRealtimeEffectUIData(*waveTrack);
+            break;
+         case TrackListEvent::DELETION:
+            if (evt.mExtra == 0)
+               HideRealtimeUIForTrack(*waveTrack);
```

Comment:
The problems of dangling pointers to states or parts of states are surely fixed now, so no fear.

Maybe though, this event handler closes the dialog too often because it isn't discriminating about which track.

But to do that right might require a shared pointer to the interesting track, not weak.

And then there would be the problem of invalidating that shared pointer, and updating it to find the corresponding track in a different undo state.

Because I observe that Undo or Redo causes the dialog to disappear, because of this case of the switch.  Maybe that is acceptable for now.  Maybe testing will result in some request to change the design.

Reply:
I do not see your point. `mExtra == 0` means that the track is removed from the project as per contract. Please, provide an STR of a behavior you think is wrong.

Reply:
Seems that undo is calling Clear, readding the duplicate back later